DataDefinitions/Ship.cs

```diff
@@ -448,15 +448,12 @@ public string SpokenModel()
         public decimal? Distance(decimal? fromX, decimal? fromY, decimal? fromZ)
         {
             // Work out the distance to the system where the ship is stored if we can
-            if (x is null || y is null || z is null || fromX is null || fromY is null || fromZ is null)
-            {
-                // We don't know how far away the ship is
-                return null;
+            if (x != null && y != null && z != null && fromX != null && fromY != null && fromZ != null) 
```

Comment:
Is there merit in pushing the null-checking into `Functions.DistanceFromCoordinates()` itself?

Reply:
Ok. Update pushed.